panic: non-positive interval for NewTicker

Sometimes the backend crashes soon after starting up with the following trace:

```
{"component":"monitor","level":"info","msg":"monitor timed out, for /sensu.io/monitors/default/keepalived/fc3cca518056, handling failure","time":"2018-12-14T22:18:41-08:00"}
{"component":"keepalived","entity":"fc3cca518056","level":"info","msg":"keepalive timed out, creating keepalive event for entity","time":"2018-12-14T22:18:41-08:00"}
panic: non-positive interval for NewTicker

goroutine 265 [running]:
time.NewTicker(0x0, 0xc420a3cb40)
	/usr/lib/go/src/time/tick.go:23 +0x1a1
github.com/sensu/sensu-go/backend/monitor.(*EtcdSupervisor).Monitor.func1(0x18fbb60, 0xc420d780c0, 0x0)
	/home/cyril/go/src/github.com/sensu/sensu-go/backend/monitor/supervisor.go:140 +0x115
created by github.com/sensu/sensu-go/backend/monitor.watchMon.func1
	/home/cyril/go/src/github.com/sensu/sensu-go/backend/monitor/supervisor.go:206 +0x1a8
[etcd-snapshot.zip](https://github.com/sensu/sensu-go/files/2687114/etcd-snapshot.zip)


```

I suspect this is due to previous entity/keepalive state in my local etcd.
This is concerning because after GA, I do not expect users to potentially have to completely erase their etcd store between sensu version updates.

## Expected Behavior

No panic.

## Steps to Reproduce (for bugs)
1. (maybe optional) unzip load the attached etcd state with `etcdctl snapshot restore`
2. start the backend
3. if the panic doesn't occur within a few seconds, stop the backend and try to start it again. It can take 10+ tries to trigger

## Your Environment

* Sensu version: c4373a6aece87679d225829160c6fd19fb14c20a
* Installation method: git
* Operating System and version: Arch Linux, kernel 4.19.4
* etcd snapshot: [etcd-snapshot.zip](https://github.com/sensu/sensu-go/files/2687128/etcd-snapshot.zip)

--- a/backend/monitor/monitor_test.go
+++ b/backend/monitor/monitor_test.go
@@ -153,7 +153,7 @@ func TestWatchMonDelete(t *testing.T) {
 
 	failWait := &sync.WaitGroup{}
 
-	testFailureHandler := func(context.Context, int64) {
+	testFailureHandler := func(context.Context) {
 		failWait.Done()
 	}
 

--- a/backend/monitor/supervisor.go
+++ b/backend/monitor/supervisor.go
@@ -12,6 +12,7 @@ import (
 
 	"github.com/coreos/etcd/clientv3"
 	"github.com/coreos/etcd/mvcc/mvccpb"
+	"github.com/sensu/sensu-go/agent"
 	corev2 "github.com/sensu/sensu-go/api/core/v2"
 	"github.com/sensu/sensu-go/backend/store"
 )
@@ -130,14 +131,24 @@ func (m *EtcdSupervisor) Monitor(ctx context.Context, name string, event *corev2
 		return err
 	}
 
-	failureFunc := func(ctx context.Context, ttl int64) {
+	failureFunc := func(ctx context.Context) {
+		interval := int64(0)
+		if event.HasCheck() {
+			// This should be the case most of the time
+			interval = int64(event.Check.Interval)
+		}
+		if interval == 0 {
+			// This can happen when monitoring an event that does not have a
+			// check, an unexpected but possible scenario.
+			interval = agent.DefaultKeepaliveInterval
+		}
 		logger.Infof("monitor timed out, for %s, handling failure", key)
 		err := m.failureHandler.HandleFailure(event)
 		if err != nil {
 			m.errorHandler.HandleError(err)
 		}
 		// Emit a failing keepalive event until the context is canceled
-		ticker := time.NewTicker(time.Duration(ttl) * time.Second)
+		ticker := time.NewTicker(time.Duration(interval) * time.Second)
 		defer ticker.Stop()
 		for {
 			select {
@@ -189,7 +200,7 @@ func (m *EtcdSupervisor) getMonitor(ctx context.Context, key string) (*monitor, 
 // watchMon takes a monitor key and watches for etcd ops. If a DELETE event
 // is witnessed, it calls the provided HandleFailure func. If a PUT event is
 // witnessed, the watcher is stopped.
-func watchMon(ctx context.Context, cli *clientv3.Client, mon *monitor, failureHandler func(context.Context, int64), shutdownHandler func()) {
+func watchMon(ctx context.Context, cli *clientv3.Client, mon *monitor, failureHandler func(context.Context), shutdownHandler func()) {
 	ctx, cancel := context.WithCancel(ctx)
 	responseChan := cli.Watch(ctx, mon.key)
 	leasesMu.Lock()
@@ -203,7 +214,7 @@ func watchMon(ctx context.Context, cli *clientv3.Client, mon *monitor, failureHa
 			case wresp := <-responseChan:
 				for _, ev := range wresp.Events {
 					if ev.Type == mvccpb.DELETE {
-						go failureHandler(ctx, mon.ttl)
+						go failureHandler(ctx)
 					}
 
 					// if there is a PUT on the key, the lease has been extended,

